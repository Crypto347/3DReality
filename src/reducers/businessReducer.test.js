// import configureStore from 'redux-mock-store';
// import * as actionTypes from "../constants/actionTypes";

// import reducer from './businessReducer';
// import {initialState} from './businessReducer'

// const mockStore = configureStore();
// const store = mockStore();

// describe('businessReducer', () => {

//     beforeEach(() => { // Runs before each test in the suite
//         store.clearActions();
//     });

//     // it("should return the initial state", () => {
//     //    expect(reducer(undefined, {})).toEqual(state);
//     // })

//     // it('should return the initial state', () => {
//     //     expect(reducer(undefined, {})).toMatchSnapshot()
//     //   })

//     // it('should handle MAKE_PAPERCLIP', () => {
//     //     expect(reducer(initialState,{type: 'MAKE_PAPERCLIP'})).toMatchSnapshot()
//     // })

//     // it("should add 1 paperclips, add 1 to unsoldInventory and subtract 1 from wire", () => {
//     //     expect(reducer(initialState, {type: actionTypes.MAKE_PAPERCLIP})).toEqual(
//     //         {
//     //            ...initialState,
//     //            paperClips: 1,
//     //            unsoldInventory: 1,
//     //            wire: 49
//     //         }
//     //        );
//     //  })


// });
