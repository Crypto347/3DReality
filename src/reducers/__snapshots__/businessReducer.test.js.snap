// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`businessReducer should handle MAKE_PAPERCLIP 1`] = `
Object {
  "autoClipperInitPrice": 0,
  "autoClipperOn": false,
  "autoClipperPrice": 6.1,
  "autoClippersButtonDisabled": true,
  "autoClippersPerSec": 0,
  "autoPaperclips": 0,
  "autoWireBuyerIsOn": true,
  "cards": Array [],
  "changedToQOps": false,
  "chips": Array [],
  "chosenListDropdown": Array [
    "Low Risk",
    "Pick a Start",
  ],
  "clipsPerSec": 0,
  "clipsToBuyTrust": 3000,
  "comments": Array [
    "Welcome to Universal Paperclips",
  ],
  "creativity": 0,
  "creativityTurnOn": false,
  "currentQOps": -360,
  "delay": 1000,
  "delayAutoPaperClippers": 1000,
  "delayUnsoldInventary": 4375,
  "delayUnsoldInventaryConst": 700,
  "delayUpdatingInvLines": 5000,
  "delayUpdatingInvScreen": 10000,
  "fakeInvestmentsCash": 0,
  "funds": 100,
  "investmentsCash": 0,
  "investmentsLines": Array [
    "",
    "",
    "",
    "",
    "",
  ],
  "investmentsStocks": 0,
  "investmentsTotal": 0,
  "listInvestments": Array [
    "Low Risk",
    "Med Risk",
    "High Risk",
  ],
  "listStrategicModeling": Array [
    "Pick a Start",
    "RANDOM",
  ],
  "makePaperclipDisabled": false,
  "marketingButtonDisabled": true,
  "marketingCost": 100,
  "marketingLevel": 1,
  "maxPublicDemand": 800,
  "megaClippersIsShown": false,
  "noWire": false,
  "ops": 0,
  "opsMax": 10,
  "paperClips": 1,
  "paperclipPrice": 1000000000,
  "prevTrust": 3000,
  "processorsMemory": 1,
  "processorsNumber": 1,
  "publicDemand": 16,
  "revTracker": false,
  "risk": "lowRisk",
  "showChip": false,
  "showDropdownInvestments": false,
  "showDropdownStrategicModeling": false,
  "showInvestmentEngine": true,
  "showQCompMessage": 0,
  "showQuantumComputing": false,
  "showStrategicModeling": true,
  "time": 0,
  "trust": 2,
  "unsoldInventory": 1,
  "wire": 49,
  "wireButtonDisabled": true,
  "wireBuyerIsShown": false,
  "wireBuyerProjectIsShown": false,
  "wirePrice": 20,
  "wireToAdd": 1000,
}
`;

exports[`businessReducer should return the initial state 1`] = `
Object {
  "autoClipperInitPrice": 0,
  "autoClipperOn": false,
  "autoClipperPrice": 6.1,
  "autoClippersButtonDisabled": true,
  "autoClippersPerSec": 0,
  "autoPaperclips": 0,
  "autoWireBuyerIsOn": true,
  "cards": Array [],
  "changedToQOps": false,
  "chips": Array [],
  "chosenListDropdown": Array [
    "Low Risk",
    "Pick a Start",
  ],
  "clipsPerSec": 0,
  "clipsToBuyTrust": 3000,
  "comments": Array [
    "Welcome to Universal Paperclips",
  ],
  "creativity": 0,
  "creativityTurnOn": false,
  "currentQOps": -360,
  "delay": 1000,
  "delayAutoPaperClippers": 1000,
  "delayUnsoldInventary": 4375,
  "delayUnsoldInventaryConst": 700,
  "delayUpdatingInvLines": 5000,
  "delayUpdatingInvScreen": 10000,
  "fakeInvestmentsCash": 0,
  "funds": 100,
  "investmentsCash": 0,
  "investmentsLines": Array [
    "",
    "",
    "",
    "",
    "",
  ],
  "investmentsStocks": 0,
  "investmentsTotal": 0,
  "listInvestments": Array [
    "Low Risk",
    "Med Risk",
    "High Risk",
  ],
  "listStrategicModeling": Array [
    "Pick a Start",
    "RANDOM",
  ],
  "makePaperclipDisabled": false,
  "marketingButtonDisabled": true,
  "marketingCost": 100,
  "marketingLevel": 1,
  "maxPublicDemand": 800,
  "megaClippersIsShown": false,
  "noWire": false,
  "ops": 0,
  "opsMax": 10,
  "paperClips": 0,
  "paperclipPrice": 1000000000,
  "prevTrust": 3000,
  "processorsMemory": 1,
  "processorsNumber": 1,
  "publicDemand": 16,
  "revTracker": false,
  "risk": "lowRisk",
  "showChip": false,
  "showDropdownInvestments": false,
  "showDropdownStrategicModeling": false,
  "showInvestmentEngine": true,
  "showQCompMessage": 0,
  "showQuantumComputing": false,
  "showStrategicModeling": true,
  "time": 0,
  "trust": 2,
  "unsoldInventory": 0,
  "wire": 50,
  "wireButtonDisabled": true,
  "wireBuyerIsShown": false,
  "wireBuyerProjectIsShown": false,
  "wirePrice": 20,
  "wireToAdd": 1000,
}
`;
